File: backend/src/Logitar.Demo.Core/Aggregate.cs
using Logitar.Demo.Core.Actors;

namespace Logitar.Demo.Core;

public abstract record Aggregate
{
  public Actor CreatedBy { get; set; } = new();
  public DateTime CreatedOn { get; set; }

  public Actor UpdatedBy { get; set; } = new();
  public DateTime UpdatedOn { get; set; }

  public long Version { get; set; }
}

File: src/types/aggregate.ts
export type Actor = {
  id: string;
  type: ActorType;
  isDeleted: boolean;
  displayName: string;
  emailAddress?: string;
  pictureUrl?: string;
};

export type ActorType = "ApiKey" | "System" | "User";

export type Aggregate = {
  id: string;
  version: number;
  createdBy: Actor;
  createdOn: string;
  updatedBy: Actor;
  updatedOn: string;
};
